text/0008-uniform-random-tip-selection/0008-uniform-random-tip-selection.md

```diff
@@ -1,206 +1,156 @@
+in the past, as such cone is likely to be already confirmed and therefore does not contribute to the
+rate of newly confirmed transactions when a milestone is issued.
+
+# Detailed design
+
+Definitions:
+* `Direct Approvers` are the set of transactions which directly approve a given transaction.
+* `Approvee` is the directly approved transaction of a given transaction.
+Example: the trunk/branch transactions are the approvees of a given transaction.
+* `Solid` means that the past cone of a given transaction exists in the database.
+* A `tail transaction` is the transaction at index zero of a bundle. Only a tail transaction can be a `tip`.
+* A `valid bundle` is a bundle which is structurally correct and has valid signatures (in case it moves funds).
+* A `tip` is a solid tail transaction of a valid bundle without any approvers. Its past cone contains only valid bundles.
+* A `score` is a scoring  determining the likeliness to select a given `tip`.
+* `Confirmed Root Transactions` defines the set of first seen transactions which are confirmed by a previous milestone 
+when we walk the past cone of a given transaction. The walk stops on confirmed transactions.  
+Note that the red marked milestone is also a `Confirmed Root Transaction`.
+![sdf](images/cnf_tx_roots.PNG)
+* `Transaction Snapshot Index (TSI)` defines the index of the milestone which confirmed a given transaction.
+* `Oldest Transaction Root Snapshot Index (OTRSI)` defines the lowest milestone index of a set of
+`Confirmed Root Transactions` of a given transaction.
+* `Youngest Transaction Root Snapshot Index (YTRSI)` defines the highest milestone index of a set of
+`Confirmed Root Transactions` of a given transaction.
+* `Latest Solid Milestone Index (LSMI)` the index of the latest solid milestone.
+* `Below Max Depth (BMD)` defines a threshold value up on which it is decided on whether a transaction is not
+relevant in relation to the recent parts of the Tangle. The current `BMD` for mainnet nodes is 15 milestones, 
+which means that transactions of which their `OTRSI` in relation to the `LSMI` is more than 15, are "below max depth".
+
+### OTRSI / YTRSI example
+Given the blue PoV transaction, the `OTRSI` of it is milestone 1 and `YTRSI` milestone 2.
+Note that here again, the milestones are also `Confirmed Root Transactions`.
+![sdf](images/otrsi_ytrsi.PNG)
+
+### Milestone based tip scoring
+
+The milestone based scoring defines a tip's score by investigating the tip's relation to
+the cone it approves and previous issued  milestones.
+
+A tip can have one of 3 score states:
+* `0`: The tip is lazy and should not be selected.
+* `1`: The tip is somewhat lazy.
+* `2`: The tip is a non-lazy tip.
+
+Definitions:
+* `C1`: Max allowed delta value for the `YTRSI` of a given transaction in relation to the current `LSMI`.
+* `C2`: Max allowed delta value between `OTRSI` of the approvees of a given transaction in relation to the current `LSMI`. 
```

Comment:
```suggestion
* `C2`: Max allowed delta value between `OTRSI` of a given transaction in relation to the current `LSMI`. 
```
We don't calculate the OTRSI of the TX's approvees anymore, but of the TX itself.